modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -216,14 +208,14 @@ private void createServerBootstrap(ProfileSettings profileSettings, NioEventLoop
 
         serverBootstrap.group(eventLoopGroup);
 
-        // If direct buffer pooling is disabled, use the CopyBytesServerSocketChannel which will create child
-        // channels of type CopyBytesSocketChannel. CopyBytesSocketChannel pool a single direct buffer
-        // per-event-loop thread to be used for IO operations.
-        if (ByteBufAllocator.DEFAULT.isDirectBufferPooled()) {
-            serverBootstrap.channel(NioServerSocketChannel.class);
-        } else {
-            serverBootstrap.channel(CopyBytesServerSocketChannel.class);
-        }
+        // CopyBytesServerSocketChannel which will create child channels of type CopyBytesSocketChannel.
+        // CopyBytesSocketChannel pool a single direct buffer per-event-loop thread to be used for IO
+        // operations.
+        serverBootstrap.channel(CopyBytesServerSocketChannel.class);
```

Comment:
Why can we unconditionally set this now? Shouldn't we check with the `NettyAllocator.ALLOCATOR`?